e2e_playwright/st_page_link_test.py

```diff
@@ -29,51 +30,66 @@ def test_page_links(app: Page, assert_snapshot: ImageCompareFunction):
     assert_snapshot(page_link_links.nth(4), name="st_page_link-sidebar-disabled")
 
 
-def test_default_container_width(app: Page, assert_snapshot: ImageCompareFunction):
-    """Test that st.page_link default container width in main is false and in sidebar is true."""
-    page_links = app.get_by_test_id("stPageLink")
-
-    page_links.nth(0).get_by_test_id("stMarkdownContainer").hover()
-    assert_snapshot(page_links.nth(0), name="st_page_link-sidebar-default")
-
-    page_links.nth(4).get_by_test_id("stMarkdownContainer").hover()
-    assert_snapshot(
-        page_links.nth(4), name="st_page_link-sidebar-container-width-false"
-    )
-
-    page_links.nth(5).get_by_test_id("stMarkdownContainer").hover()
-    assert_snapshot(page_links.nth(5), name="st_page_link-default")
-
-    page_links.nth(9).get_by_test_id("stMarkdownContainer").hover()
-    assert_snapshot(page_links.nth(9), name="st_page_link-container-width-true")
-
-
 def test_page_link_help_tooltip(app: Page):
     """Test that st.page_link help tooltip renders correctly."""
     page_links = app.get_by_test_id("stPageLink")
     expect(page_links).to_have_count(PAGE_LINK_COUNT)
 
     # Get the tooltip hover target and ensure it's visible before hovering
-    hover_target = page_links.nth(7).get_by_test_id("stTooltipHoverTarget")
+    hover_target = page_links.nth(9).get_by_test_id("stTooltipHoverTarget")
     expect(hover_target).to_be_visible()
 
     # Hover over the tooltip target
     hover_target.hover()
 
     expect(app.get_by_text("Some help text")).to_be_visible()
+
+
+def test_page_link_width_examples(app: Page, assert_snapshot: ImageCompareFunction):
+    """Test page link width examples via screenshot matching."""
+    page_expander = get_expander(app, "Page Link Width Examples")
+
+    page_elements = page_expander.get_by_test_id("stPageLink")
+
+    assert_snapshot(page_elements.nth(0), name="st_page_link-width_content")
+    assert_snapshot(page_elements.nth(1), name="st_page_link-width_stretch")
+    assert_snapshot(page_elements.nth(2), name="st_page_link-width_500px")
+    assert_snapshot(page_elements.nth(3), name="st_page_link-width_content_help")
+    assert_snapshot(page_elements.nth(4), name="st_page_link-width_stretch_help")
+    assert_snapshot(page_elements.nth(5), name="st_page_link-width_500px_help")
+
+    # Test gradual deprecation of use_container_width
+    assert_snapshot(
+        get_element_by_key(app, "main_container_width_true"),
+        name="st_page_link-main_container_width_true",
+    )
+    assert_snapshot(
+        get_element_by_key(app, "main_container_width_false"),
+        name="st_page_link-main_container_width_false",
+    )
+    assert_snapshot(
+        get_element_by_key(app, "sidebar_container_width_true"),
+        name="st_page_link-sidebar_container_width_true",
+    )
+    assert_snapshot(
+        get_element_by_key(app, "sidebar_container_width_false"),
+        name="st_page_link-sidebar_container_width_false",
+    )
```

Comment:
Depending on how we proceed with the deprecation warning, it would be better to just check that the warning is displayed instead of doing full snapshot tests.